File: cpp/Dpr.cpp
/*Saya Muhammad Muhammad Fadlul Hafiizh [2209889] mengerjakan soal latprak_1 dalam mata kuliah DPBO.
untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan, Aamiin */

#include <iostream>
#include <string>

using namespace std;

class Dpr{
    private:
        int id;
        string nama;
        string nama_bidang;
        string nama_partai;

    public:
        //constructor sebagai nilai default saat instansiasi
        Dpr(){
            this->nama = "-";
            this->nama_bidang = "-";
            this->nama_partai = "-";
        }

        //enkapsulasi semua atribut
        int getId()
        {
            return this->id;
        }

        void setId(int id)
        {
            this->id = id;
        }

        string getNama()
        {
            return this->nama;
        }

        void setNama(string nama)
        {
            this->nama = nama;
        }

        string getNama_bidang()
        {
            return this->nama_bidang;
        }

        void setNama_bidang(string nama_bidang)
        {
            this->nama_bidang = nama_bidang;
        }

        string getNama_partai()
        {
            return this->nama_partai;
        }

        void setNama_partai(string nama_partai)
        {
            this->nama_partai = nama_partai;
        }

        // destructor
        ~Dpr(){}
        
};

File: python/Dpr.py
#Saya Muhammad Muhammad Fadlul Hafiizh [2209889] mengerjakan soal latprak_1 dalam mata kuliah DPBO.
#untuk keberkahanNya maka saya tidak melakukan kecurangan seperti yang telah dispesifikasikan, Aamiin

import uuid
class Dpr:
    #ATRIBUT
    __id = 0
    __nama = ''
    __bidang = ''
    __partai = ''

    #constructor
    def __init__(self,nama="", bidang="", partai=""):
        self.__id = uuid.uuid4()
        self.__nama = nama
        self.__bidang = bidang
        self.__partai = partai

    #enkapsulasi setiap atribut
    def get_id(self):
        return self.__id
    
    def set_id(self, value):
        self.__id = value

    def get_nama(self):
        return self.__nama
    
    def set_nama(self, value):
        self.__nama = value

    def get_bidang(self):
        return self.__bidang
    
    def set_bidang(self, value):
        self.__bidang = value
        
    def get_partai(self):
        return self.__partai
    
    def set_partai(self, value):
        self.__partai = value